esmvalcore/cmor/_fixes/cordex/project.py

```diff
@@ -0,0 +1,70 @@
+"""Fixes for CORDEX project"""
+import numpy as np
+from scipy.interpolate import InterpolatedUnivariateSpline
+from scipy.ndimage import map_coordinates
+
+from esmvalcore.cmor.fix import Fix
+
+
+class AllVars(Fix):
+    """Generic fixes for the CORDEX project"""
+
+    def fix_metadata(self, cubes):
+        """Fix metatdata.
+
+        Calculate missing latitude/longitude boundaries using interpolation.
+
+        Parameters
+        ----------
+        cubes: iris.cube.CubeList
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            if not cube.coords('latitude'):
+                continue
+            if not cube.coords('longitude'):
+                continue
+            if cube.coord('latitude').has_bounds() and \
+                    cube.coord('longitude').has_bounds():
+                continue
+            rlat = cube.coord('grid_latitude').points
+            rlon = cube.coord('grid_longitude').points
+
+            # Transform grid latitude/longitude to array indices [0, 1, 2, ...]
+            rlat_to_idx = InterpolatedUnivariateSpline(
+                rlat, np.arange(len(rlat)), k=1)
+            rlon_to_idx = InterpolatedUnivariateSpline(
+                rlon, np.arange(len(rlon)), k=1)
+            rlat_idx_bnds = rlat_to_idx(
+                cube.coord('grid_latitude')._guess_bounds())
```

Comment:
It would be better if you could avoid using private, undocumented functions, as these may change or be removed at any time without warning.

Reply:
Done

Reply:
Thanks!